### PR TITLE
fix: blocking emeritus product source

### PR DESCRIPTION
**Pull request by kiram15**
## Description

Adding in the allowed values for the new `product_source` field, purposefully excluding Emeritus courses. 

## Ticket Link

Link to the associated ticket
[ENT-6785](https://2u-internal.atlassian.net/browse/ENT-6785)

## Post-review

* [ ] Squash commits into discrete sets of changes
* [ ] Ensure that once the changes have been [deployed to stage](https://gocd.tools.edx.org/go/pipeline/activity/stage-enterprise_catalog), prod is manually deployed

### CODE REVIEW

File: enterprise_catalog/apps/catalog/models.py [hunk 1]
```diff
@@ -732,13 +733,15 @@ def _should_allow_metadata(metadata_entry, catalog_query=None):
     """
     # make sure to exclude exec ed course runs
     content_type = get_content_type(metadata_entry)
+    entry_product_source = metadata_entry.get('product_source')
```
Comment (bcitro) [inline enterprise_catalog/apps/catalog/models.py]: I think we'll want to write a helper function (something like `get_product_source(metadata_entry)`) to extract the product source from the metadata, since it may be in different formats depending on the discovery endpoint used. The `search/all` endpoint will be a string (e.g. `"product_source": "2u"`), but `courses` endpoint will be a dictionary, e.g.:

`"product_source": {
        "name": "2u",
        "slug": "2u",
        "description": "2U, Trilogy, Getsmarter -- external source for 2u courses and programs"
    }`

File: enterprise_catalog/apps/catalog/tests/test_models.py [hunk 2]
```diff
@@ -85,6 +85,46 @@ def test_2U_exec_ed_content_inclusion_logic(self, mock_client):
...
+            ('aggregation_key', 'course:edX+edxSourceX'),
+            ('key', 'edX+edxSourceX'),
+            ('title', 'test course'),
+            ('product_source', 'edX'),
+        ])
+        twou_course_metadata = OrderedDict([
+            ('aggregation_key', 'course:edX+2uSourceX'),
+            ('key', 'edX+2uSourceX'),
+            ('title', 'test course 2'),
+            ('product_source', OrderedDict(
```
Comment (johnnagro) [inline enterprise_catalog/apps/catalog/tests/test_models.py:105]: great that you have both the string and dict style metadata in your test 👍

File: enterprise_catalog/apps/catalog/models.py [hunk 3]
```diff
@@ -730,6 +739,9 @@ def _should_allow_metadata(metadata_entry, catalog_query=None):
     Returns:
         bool: If we should save the metadata as a ContentMetaData object
     """
+    entry_product_source = _fetch_product_source(metadata_entry)
+    if entry_product_source not in CONTENT_PRODUCT_SOURCE_ALLOW_LIST and entry_product_source is not None:
```
Comment (johnnagro) [inline enterprise_catalog/apps/catalog/models.py:743]: we may find that we need this to be case insensitive in the future.

**Review by johnnagro — APPROVED** [review]
⛵

File: enterprise_catalog/apps/catalog/tests/test_models.py [hunk 4]
```diff
@@ -85,6 +85,46 @@ def test_2U_exec_ed_content_inclusion_logic(self, mock_client):
         update_contentmetadata_from_discovery(catalog.catalog_query)
         self.assertEqual(ContentMetadata.objects.count(), 3)
 
+    @override_settings(DISCOVERY_CATALOG_QUERY_CACHE_TIMEOUT=0)
+    @mock.patch('enterprise_catalog.apps.api_client.discovery_cache.DiscoveryApiClient')
+    def test_product_source_content_inclusion_logic(self, mock_client):
```
Comment (bcitro) [inline enterprise_catalog/apps/catalog/tests/test_models.py:90]: can we add a test case for null product_source?

File: enterprise_catalog/apps/catalog/models.py (hunk 3 above)
Comment (alex-sheehan-edx) [inline enterprise_catalog/apps/catalog/models.py:743]: tiny nit- I believe it's possible to shorten this to `if entry_product_source and entry_product_source not in CONTENT_PRODUCT_SOURCE_ALLOW_LIST:`

**Review by alex-sheehan-edx — APPROVED** [review]
tiny nit but not requiring a change 👍
